core/Azure.Mcp.Core/src/Areas/Server/Commands/ServiceStartCommand.cs

```diff
@@ -134,6 +169,55 @@ private IHost CreateHost(ServiceStartOptions serverOptions)
             .Build();
     }
 
+    /// <summary>
+    /// Creates a host for HTTP transport.
+    /// </summary>
+    /// <param name="serverOptions">The server configuration options.</param>
+    /// <returns>An IHost instance configured for HTTP transport.</returns>
+    private IHost CreateHttpHost(ServiceStartOptions serverOptions)
+    {
+        return Host.CreateDefaultBuilder()
+            .ConfigureLogging(logging =>
+            {
+                logging.ClearProviders();
+                logging.ConfigureOpenTelemetryLogger();
+                logging.AddEventSourceLogger();
+                logging.AddConsole();
+            })
+            .ConfigureWebHostDefaults(webBuilder =>
+            {
+                webBuilder.ConfigureServices(services =>
+                {
+                    services.AddCors(options =>
+                    {
+                        options.AddPolicy("AllowAll", policy =>
+                        {
+                            policy.AllowAnyOrigin()
+                                  .AllowAnyMethod()
+                                  .AllowAnyHeader();
+                        });
+                    });
+
+                    ConfigureServices(services);
+                    ConfigureMcpServer(services, serverOptions);
+                });
+
+                webBuilder.Configure(app =>
+                {
+                    app.UseCors("AllowAll");
+                    app.UseRouting();
+                    app.UseEndpoints(endpoints =>
+                    {
+                        endpoints.MapMcp();
+                    });
+                });
+
+                var url = Environment.GetEnvironmentVariable("ASPNETCORE_URLS") ?? "http://localhost:5001";
```

Comment:
You are using the commplex-link protocol port, is that safe? 

How do you ensure that `ASPNETCORE_URLS` only binds to safe IP addresses?

Reply:
thanks Larry, updated the pr, posting what we discussed internally

Binding to 0.0.0.0 is a requirement for containerized deployments (Docker, AKS), since loopback-only (127.0.0.1) would make the service unreachable outside the container. At the same time, 0.0.0.0 can expose the server to unintended access. To balance both needs, we are introducing an explicit opt-in flag (e.g., ALLOW_INSECURE_EXTERNAL_BINDING=true) that allows binding to 0.0.0.0. This way, Foundry and other internal partners deploying in Docker/AKS can enable it as required, while default behavior stays safe and we warn or throw binding to loop back unless an explicit opt-in flag